libass/ass_render.c

```diff
@@ -1850,40 +1851,42 @@ fix_glyph_scaling(ASS_Renderer *priv, GlyphInfo *glyph)
     glyph->font_size = ft_size;
 }
 
- /**
-  * \brief Checks whether a glyph should start a new bitmap run
-  * \param info Pointer to new GlyphInfo to check
-  * \param current_info Pointer to CombinedBitmapInfo for current run (may be NULL)
-  * \return true if a new run should be started
-  */
-static bool is_new_bm_run(GlyphInfo *info, GlyphInfo *last)
+// Initial run splitting based purely on the characters' styles
+static void split_style_runs(ASS_Renderer *render_priv)
 {
-    return !last || info->effect || info->drawing_text || last->drawing_text ||
-        strcmp(last->font->desc.family, info->font->desc.family) ||
-        last->font->desc.vertical != info->font->desc.vertical ||
-        last->font_size != info->font_size ||
-        last->c[0] != info->c[0] ||
-        last->c[1] != info->c[1] ||
-        last->c[2] != info->c[2] ||
-        last->c[3] != info->c[3] ||
-        last->be != info->be ||
-        last->blur != info->blur ||
-        last->shadow_x != info->shadow_x ||
-        last->shadow_y != info->shadow_y ||
-        last->frx != info->frx ||
-        last->fry != info->fry ||
-        last->frz != info->frz ||
-        last->fax != info->fax ||
-        last->fay != info->fay ||
-        last->scale_x != info->scale_x ||
-        last->scale_y != info->scale_y ||
-        last->border_style != info->border_style ||
-        last->border_x != info->border_x ||
-        last->border_y != info->border_y ||
-        last->hspacing != info->hspacing ||
-        last->italic != info->italic ||
-        last->bold != info->bold ||
-        ((last->flags ^ info->flags) & ~DECO_ROTATE);
+    for (int i = 1; i < render_priv->text_info.length; i++) {
+        GlyphInfo *info = render_priv->text_info.glyphs + i;
+        GlyphInfo *last = render_priv->text_info.glyphs + (i - 1);
```

Comment:
I think that `last` is misleading here, shouldn't it be `prev` or something?

Reply:
Well, “last” can also mean “prev”, and here I tried to keep as much code unmodified as I could, barring the indentation change. We use `last...` for “most recent X” in a few places, although some of them skip some elements so their “most recent” is not always the immediately preceding element. But e. g. in `ass_shaper_find_runs` it actually is.